contracts/poolMaster/PoolMaster.sol

```diff
@@ -0,0 +1,298 @@
+    _deposit();
+  }
+
+  /*
+   * @notice Called by users burning their token
+   * @dev Calculates pro rata KNC and redeems from staking contract
+   * @param tokensToRedeem
+   */
+  function withdraw(uint256 tokensToRedeemTwei) external nonReentrant {
+    require(balanceOf(msg.sender) >= tokensToRedeemTwei, 'insufficient balance');
+
+    uint256 proRataKnc = getLatestStake().mul(tokensToRedeemTwei).div(totalSupply());
+    _unstake(proRataKnc);
+    super._burn(msg.sender, tokensToRedeemTwei);
+
+    newKnc.safeTransfer(msg.sender, proRataKnc);
+  }
+
+  /*
+   * @notice Vote on KyberDAO campaigns
+   * @dev Admin calls with relevant params for each campaign in an epoch
+   * @param proposalId: DAO proposalId
+   * @param optionBitMask: voting option
+   */
+  function vote(uint256 proposalId, uint256 optionBitMask) external onlyAdminOrOperator {
+    kyberGovernance.submitVote(proposalId, optionBitMask);
+  }
+
+  /*
+   * @notice Claim accumulated reward thus far
+   * @notice Will apply admin fee to KNC token.
+   * Admin fee for other tokens applied after liquidation to KNC
+   * @dev Admin or operator calls with relevant params
+   * @param cycle - sourced from Kyber API
+   * @param index - sourced from Kyber API
+   * @param tokens - ERC20 fee tokens
+   * @param merkleProof - sourced from Kyber API
+   */
+  function claimReward(
+    uint256 cycle,
+    uint256 index,
+    IERC20Ext[] calldata tokens,
+    uint256[] calldata cumulativeAmounts,
+    bytes32[] calldata merkleProof
+  ) external onlyAdminOrOperator {
+    rewardsDistributor.claim(
+      cycle,
+      index,
+      address(this),
+      tokens,
+      cumulativeAmounts,
+      merkleProof
+    );
+
+    for (uint256 i = 0; i < tokens.length; i++) {
+      if (tokens[i] == newKnc) {
+        uint256 availableKnc = _administerAdminFee(getAvailableNewKncBalanceTwei());
+        _stake(availableKnc);
+      }
+    }
+  }
+
+  /*
+   * @notice Will liquidate ETH or ERC20 tokens to KNC
+   * @notice Will apply admin fee after liquidations
+   * @dev Admin or operator calls with relevant params
+   * @param tokens - ETH / ERC20 tokens to be liquidated to KNC
+   * @param minRates - kyberProxy.getExpectedRate(eth/token => knc)
+  */
+  function liquidateTokensToKnc(IERC20Ext[] calldata tokens, uint256[] calldata minRates)
+    external
+    onlyAdminOrOperator
+  {
+    require(tokens.length == minRates.length, 'unequal lengths');
+    for (uint256 i = 0; i < tokens.length; i++) {
+      if (tokens[i] == ETH_ADDRESS) {
+        kyberProxy.swapEtherToToken{value: address(this).balance}(newKnc, minRates[i]);
+      } else if (tokens[i] != newKnc) {
+        kyberProxy.swapTokenToToken(
+          tokens[i],
+          tokens[i].balanceOf(address(this)),
+          newKnc,
+          minRates[i]
+        );
+      }
+    }
+    uint256 availableKnc = _administerAdminFee(getAvailableNewKncBalanceTwei());
+    _stake(availableKnc);
```

Comment:
Pls consider, stake all of knc and mint to owner an amount that equals to the proportion of knc in the pool

Reply:
that will mean that this pool master token will always be 1:1 KNC, and it's up to them to decide how to distribute the rewards?

Reply:
let me check with unagii and defi dude, see if they have preference

Reply:
No, it is not.
For example:

the pool has 100 knc with the total supply 1000 lp token.
Next epoch the reward is 10 knc with 5 knc for staker and 5 knc for owner.
1000 lp token is equals to 105 knc, now
5 knc for owner will be restaked and mint for owner 5 * 1000 / 105 token

Reply:
oh, means helping owners stake the admin fee?

Reply:
will add a stakeAdminFee function for operator to stake fees, since there's mint, burn and claim fees now